SU2_CFD/src/solvers/CIncNSSolver.cpp

```diff
@@ -586,3 +599,275 @@ void CIncNSSolver::BC_ConjugateHeat_Interface(CGeometry *geometry, CSolver **sol
     nodes->SetEnergy_ResTruncError_Zero(iPoint);
   }
 }
+
+
+
+
+void CIncNSSolver::SetTauWall_WF(CGeometry *geometry, CSolver **solver_container, const CConfig *config) {
+  /*---  The wall function implemented herein is based on Nichols and Nelson AIAAJ v32 n6 2004.
+   At this moment, the wall function is only available for adiabatic flows.
+   ---*/
+  su2double RefDensity,RefVel2; 
+  if ((config->GetRef_Inc_NonDim() == DIMENSIONAL) || (config->GetRef_Inc_NonDim() == INITIAL_VALUES)) {
+    RefDensity = Density_Inf;
+    RefVel2 = 0.0;
+    for (auto iDim = 0u; iDim < nDim; iDim++) 
+      RefVel2 += Velocity_Inf[iDim] * Velocity_Inf[iDim];
+  } else if (config->GetRef_Inc_NonDim() == REFERENCE_VALUES) {
+    RefDensity = config->GetInc_Density_Ref();
+    RefVel2 = config->GetInc_Velocity_Ref() * config->GetInc_Velocity_Ref();
+  }
```

Comment:
This needs to exist in only one place, I fixed a recent bug due to having this logic in 4 different functions.